vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data.cc

```diff
@@ -79,6 +80,20 @@ VectorData::VectorData(std::vector<float> data) : Data(DataType::kVector) {
       static_cast<int>(data.size()), std::vector<uint32_t>(), std::move(data));
 }
 
+VectorData::VectorData(std::string string) : Data(DataType::kVector) {
+  const std::vector<std::string> vector_string = base::SplitString(
+      string, " ", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+  std::vector<float> vector;
+  for (const std::string& element_string : vector_string) {
+    double element;
+    base::StringToDouble(element_string, &element);
+    vector.push_back(element);
+  }
```

Comment:
We should store as a vector of doubles in the data structure and only serialize/deserialize when needed, i.e. when saving/loading from the database